pages/plant/[id].js

```diff
@@ -47,10 +48,28 @@ const StyledButtonContainer = styled.div`
   `}
 `;
 
-export default function PlantDetails({ plants, onDeletePlant }) {
+const StyledEditButton = styled.button`
+  border-style: none;
+  border-radius: 50px;
+  width: 48px;
+  height: 48px;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  position: absolute;
+  right: 40px;
+  /* top: 100px; */
+`;
+
+
+export default function PlantDetails({ plants, onDeletePlant, onCreatePlant, onEditPlant }) {
   const [isDeleteOption, setIsDeleteOption] = useState(false);
   const [toggleButtonName, setToggleButtonName] = useState("Delete");
+  const [isFormVisible, setIsFormVisible] = useState(false);
 
+  function toggleFormVisibility() {
+    setIsFormVisible((prevState) => !prevState);
+  }
```

Comment:
1. Move `toggleFormVisibility` and `handleCancel` functions below the router logic. Placing them below helps keep the code organized and reduces the risk of unexpected bugs related to function execution order.

2.  Use `isFormVisible` to close the form instead of reloading the page. Replace `window.location.reload()` (from the `PlantForm.js`) with a function like `handleCancel`, which simply sets `isFormVisible` to false to close the form. Pass this function as a prop to the form to handle cancel actions gracefully:

```js
 function handleCancel() {
    setIsFormVisible(false);
  }
```